core/src/Kokkos_View.hpp

```diff
@@ -449,6 +449,98 @@ struct ViewTraits {
 template <class DataType, class... Properties>
 class View;
 
+namespace Impl {
+
+/*
+ * \class ViewTracker
+ * \brief template class to wrap the shared allocation tracker
+ *
+ * \section This class is templated on the View and provides
+ * constructors that match the view.  The constructors and assignments
+ * from view will externalize the logic needed to enable/disable
+ * ref counting to provide a single gate to enable further developments
+ * which may hing on the same logic.
+ *
+ */
+template <class ParentView>
+struct ViewTracker {
```

Comment:
Can we maybe consider putting this in it's own file? `Kokkos_View.hpp` is already *way* too long...

Reply:
added code to impl/Kokkos_ViewTracker.hpp